File: 36-valid-sudoku/valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        short Col[9]={0};
        short Row[9]={0};
        short Block[9]={0};

        for(int i=0; i<9; i++){
            for(int j=0; j<9; j++){
                char c = board[i][j];
                if (c == '.') continue;
                int x=(c-'0')%9, x2=1<<x, bidx=(i/3)*3+j/3; 
                if ((Row[i]>>x)&1||(Col[j]>>x)&1||(Block[bidx]>>x)&1) return 0;
                Row[i]|=x2, Col[j]|=x2, Block[bidx]|=x2;
            }
        }
        return 1;
    }
};

File: 36-valid-sudoku/valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
    
        row = [set() for _ in range(9)]
        col = [set() for _ in range(9)]
        box = [set() for _ in range(9)]

        for i in range(9):
            for j in range(9):
                c = board[i][j]
                if c == '.':
                    continue
                idx = (i // 3) * 3 + (j // 3)
                if c in row[i] or c in col[j] or c in box[idx]:
                    return False
                row[i].add(c)
                col[j].add(c)
                box[idx].add(c)
        return True
